lib/handler/pool/state.ex

```diff
@@ -223,8 +223,9 @@ defmodule Handler.Pool.State do
     end
   end
 
+  @foo_mfa {__MODULE__, :foo, 0}
   defp shutdown_and_cleanup(state, ref, task_pid, exception) do
-    task = %Task{ref: ref, pid: task_pid, owner: self()}
+    task = %Task{mfa: @foo_mfa, ref: ref, pid: task_pid, owner: self()}
```

Comment:
We were building a fake `Task` so that we could re-use the `Task.shutdown` implementation, but perhaps we should just copy that function into our own module?

That way we won't have to fake out the task structure and we won't be fragile to the Elixir version?

Reply:
@marcelloma had a *much* better idea for fixing this: cache the `%Task{}` object returned from `Task.spawn()` in the worker state blobs, with the obvious benefit of isolating us from the internals of the Task module.

Reply:
FWIW, this package doesn't compile against `elixir 1.10`, so I turned the minimum supported version up to eleven.